### PR TITLE
feat(linter): add Vue loader

### PR DESCRIPTION
**Pull request by mysteryven**

### CODE REVIEW

**Comment by mysteryven** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1814?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #1745** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1745?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #1760** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1760?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #1814** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1814?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/oxc-project/oxc/1814?utm_source=stack-comment).

**Comment by Dunqing** [comment]
Vue SFC supports multiple script tags. But I don't see any tests containing that usage. Can we need to support it?

**Comment by Boshen** [comment]
> Vue SFC supports multiple script tags. But I don't see any tests containing that usage. Can we need to support it?

We can add a bunch of follow up issues. Let's get the overall architecture merged so we can get more helping helps on this.

**Review by Dunqing — APPROVED** [review]

File: crates/oxc_linter/src/partial_loader/vue_partial_loader.rs [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+    fn test_brace_with_regex_in_template_literal() {
+        let source_text = r"
+        <script setup>
+            `${/{/}`
+        </script>
+        ";
+
+        let loader_value = VuePartialLoader::from(source_text).build();
+        assert_eq!(loader_value.source_text.trim(), r"`${/{/}`");
+    }
```
Comment (mysteryven) [inline crates/oxc_linter/src/partial_loader/vue_partial_loader.rs:436]: How I can solve it, I have no idea whether `/` is an operator or RegExp.

Comment (Boshen) [reply crates/oxc_linter/src/partial_loader/vue_partial_loader.rs:435]: You can try and parse it as an expression?

**Comment by Boshen** [comment]
### Merge activity

* **Dec 31, 9:14 AM**: @Boshen started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/oxc-project/oxc/1814).

**Pull request merged**